src/visitors/longitudinal_diffusion_visitor.cpp

```diff
@@ -0,0 +1,51 @@
+#include "longitudinal_diffusion_visitor.hpp"
+
+#include "ast/ast_decl.hpp"
+#include "ast/kinetic_block.hpp"
+#include "ast/longitudinal_diffusion_block.hpp"
+#include "ast/name.hpp"
+#include "ast/program.hpp"
+#include "ast/statement.hpp"
+#include "ast/statement_block.hpp"
+#include "ast/string.hpp"
+#include "visitor_utils.hpp"
+
+#include <memory>
+
+
+namespace nmodl {
+namespace visitor {
+
+static std::shared_ptr<ast::StatementBlock> make_statement_block(
+    ast::KineticBlock& kinetic_block,
+    nmodl::ast::AstNodeType node_type) {
+    auto nodes = collect_nodes(kinetic_block, {node_type});
+
+    ast::StatementVector statements;
+    statements.reserve(nodes.size());
+    for (auto& node: nodes) {
+        statements.push_back(std::dynamic_pointer_cast<ast::Statement>(node));
+    }
+
+    return std::make_shared<ast::StatementBlock>(std::move(statements));
+}
+
+
+std::shared_ptr<ast::LongitudinalDiffusionBlock> create_block(ast::KineticBlock& node) {
```

Comment:
Should this be used externally? If not, it's missing a `static` specifier. If yes, it's missing a docstring.

Reply:
Correct, can be static.